### PR TITLE
[TS]LPS-163136 Bump Tika-Core and Tika-Parsers to 1.28.5 

### PR DESCRIPTION
**Pull request by javalosjr96**
Ticket:
https://issues.liferay.com/browse/LPS-163136

Issue:
A customer has recently flagged tika-core with https://github.com/advisories/GHSA-rpjm-422r-95mh.

Solution:
Bump Tika to 1.28.5

Due to the connection with tika-parsers, tika-parsers will also have to be updated.

POI/POI-OOXML/POI-SCRATCHPAD were patched and updated.
POI-OOXML-SCHEMAS was replaced by POI-OOXML-LITE(also patched)
XMLBEANS was updated as it was listed as a dependency(also patched)

Added commons-io to batch-service-engine in order to resolve test failures introduced by updating POI/POI-OOXML.
Jasperreports also need to be updated to 6.20.0 in order  to resolve failing tests for
portal-reports-engine-console-jasper-test.

Versions.html and Versions-Complete.xml were updated manually

### CODE REVIEW

**Comment by javalosjr96** [comment]
ci:test:sf

**Comment by javalosjr96** [comment]
SF Error is unrelated.

**Comment by dantewang** [comment]
ci:stop

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
@javalosjr96 No regressions found.

**Comment by dantewang** [comment]
@javalosjr96 Got some questions.

1. This pull updates a lot of libs, are they all about tika?
2. It seems all the third-party libs you update are used by modules, do they really need the patching to remove JDK9 module info classes?

**Comment by javalosjr96** [comment]
@dantewang 
Hi Dante,
Yes,
Tika-Parsers needed to be on the same version as Tika-core which required POI(and dependencies) and XMLBeans to be updated.  After POI was updated, jasperreports also needed to be updated.  
The JDK module-info.class needed to be removed in order otherwise the bundle using it would through this error while building:
error  : Classes found in the wrong directory: {META-INF/versions/9/module-info.class=module-info}	

I followed [LPS-89886](https://issues.liferay.com/browse/LPS-89886) as an example.

**Comment by dantewang** [comment]
@javalosjr96 Thanks!

**Comment by dantewang** [comment]
@javalosjr96 Could you update the commit messages to include that information in the commit messages, so that people can easily know the POI and XMLBeans are updated because of Tika-Parsers, and JasperReports is updated because of POI? Thanks!

**Comment by javalosjr96** [comment]
@dantewang 
I updated the commit messages to be more descriptive.

**Comment by javalosjr96** [comment]
@dantewang 
Hey Dante,
Does everything look good?

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
The force push only updates commit messages, which does require retesting.

**Comment by tinatian** [comment]
@vicnate5 This pull is related to portal tika, any other tests we need to run?

**Comment by vicnate5** [comment]
ci:test:document

**Comment by vicnate5** [comment]
ci:test:reports

**Comment by vicnate5** [comment]
ci:test:depot

**Comment by vicnate5** [comment]
@Kvale1733 Can you check the additional results here?

**Comment by Kvale1733** [comment]
@javalosjr96 Regression found in a reports test CPReportsadmin#DownloadReportXLS [Testray](https://testray.liferay.com/home/-/testray/case_results/2059046465)

Steps to Reproduce:
      i. Open the product menu and go to Configuration > Reports Admin
      ii. Go to Definitions Tab
      iii. Add new Definition named "Reports Definition Name" and use this file as template: https://drive.google.com/file/d/1caWA6qetyjlc9UG_bag0jczMMa8TD_GS/view?usp=sharing. Keep data source as Portal.
      iv. Click on the ellipses next to the definition entry and go to Add Report
      v. Change report format to xls
      vi. Use test@liferay.com for email notifications and recipient
      vii. Click generate to generate the report
      viii. Click on the report entry name to view it

Expected results: Can view the report entry without error
Actual results:
```
2022-11-17 18:02:21.365 WARN  [liferay/report_request-2][POIDocument:169] DocumentSummaryInformation property set came back as null
2022-11-17 18:02:21.366 WARN  [liferay/report_request-2][POIDocument:169] SummaryInformation property set came back as null
2022-11-17 18:02:21.799 WARN  [liferay/report_request-2][Section:730] The codepage property is not set although a dictionary is present. Defaulting to ISO-8859-1.
2022-11-17 18:02:21.807 INFO  [liferay/report_request-2][POIDocument:334] Wrote property set SummaryInformation of size 152
2022-11-17 18:02:21.807 WARN  [liferay/report_request-2][Section:730] The codepage property is not set although a dictionary is present. Defaulting to ISO-8859-1.
2022-11-17 18:02:21.807 INFO  [liferay/report_request-2][POIDocument:334] Wrote property set DocumentSummaryInformation of size 56

```

Only when running the Poshi the following also appears:
```
Error on line 8792 of document  : An invalid XML character (Unicode: 0x5) was found in the CDATA section. Nested exception: An invalid XML character (Unicode: 0x5) was found in the CDATA section.
```

**Comment by javalosjr96** [comment]
@Kvale1733 
Hi Kevin,

I compared the two reports generated,  one with tika/poi/jasper reports still at their current versions and the other with updates libraries. 

They both create the reports without any issues and I uploaded them to google sheets and they are both identical it seems.

Was the report generated in your instance?

**Comment by Kvale1733** [comment]
@javalosjr96 Yes the report was generated and I could download it, but the errors were unique to this PR and caused the test to fail.

**Comment by javalosjr96** [comment]
@Kvale1733  @vicnate5 
Hi Team,
For the first warns that are being displayed,  I believe they are expected as they are thrown in master currently however I think the difference would be the way the methods are thrown and how Liferay handles the WARNS for each.


In POI 4.1.2 , specifically in Section.Class , the WARN is thrown using POI Logger.

```
        int codepage = getCodepage();
        if (codepage == -1) {
            String msg =
                "The codepage property is not set although a dictionary is present. "+
                "Defaulting to ISO-8859-1.";
            LOG.log(POILogger.WARN, msg);
            codepage = Property.DEFAULT_CODEPAGE;
        }
```
        
In POI 5.2.2, they switched to LOG:

```
        int codepage = getCodepage();
        if (codepage == -1) {
            LOG.atWarn().log("The codepage property is not set although a dictionary is present. " +
                    "Defaulting to ISO-8859-1.");
            codepage = Property.DEFAULT_CODEPAGE;
        }
```  
        
       
The Image below shows the WARN being thrown:
image
*using POI 4.1.2

**Comment by javalosjr96** [comment]
Still investigating the second error from POSHI.

**Comment by javalosjr96** [comment]
I believe that the error displayed in POSHI is also caused by the WARN being thrown, however I'm having trouble debugging Liferay Selenium Class files.  

The stacktrace:
```
     [exec]     D:\Liferay-Portal-CE\liferay-portal\modules\dxp\apps\portal-reports-engine-console\portal-reports-engine-console-test\src\testFunctional\tests\CPReportsadmin.testcase[DownloadReportXLS]:497
     [exec]             at org.dom4j.io.SAXReader.read(SAXReader.java:482)
     [exec]             at org.dom4j.io.SAXReader.read(SAXReader.java:343)
     [exec]             at com.liferay.poshi.runner.selenium.LiferaySeleniumUtil.assertConsoleErrors(LiferaySeleniumUtil.java:91)
```
Is caused by this string containing the errors caught during the POSHI test:
content = getTestConsoleLogFileContent();

**Comment by dantewang** [comment]
@zhangyan0701 Could you double check the tests and see if we have any logic to ignore these warnings that needs to be updated, according to Kevin and Jorge's comments? Thanks!

**Comment by javalosjr96** [comment]
@dantewang 
Also I've confirmed that the POSHI warning is due to the new warns being thrown. 
https://liferay.slack.com/archives/CLLGDBDQD/p1668796894841969

**Comment by javalosjr96** [comment]
I'll  rebase this to a later master.

**Comment by javalosjr96** [comment]
ci:test:sf

**Comment by javalosjr96** [comment]
SF fail is unrelated.

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
@javalosjr96 The warns still appear. Only when creating the report in xls format. Locally only the warns appear. If this is a false positive, then we can just ignore the poshi error for the test to pass. We can just add the error to portal-web/test-ignorable-error-lines.xml. Like this: https://github.com/Kvale1733/liferay-portal/pull/235/commits/cbc7a587ecd2c9a8bd539c613a02950da0d0bdee

**Comment by javalosjr96** [comment]
ci:test:relevant

**Comment by javalosjr96** [comment]
@Kvale1733 Added your commit and retesting.

**Comment by Kvale1733** [comment]
@javalosjr96 Nevermind. I don't think the error can be ignore. To do that we'd need to add a different property to the test that would assert all warnings. And that test gets various warnings often so it would still fill frequently.

**Comment by javalosjr96** [comment]
@Kvale1733 
So what would need to be done?
I can patch that file to remove that WARN.

Can this PR moved forward atleast?

**Comment by Kvale1733** [comment]
@javalosjr96.You can remove my commit to ignore the error since it won't work. The error message the test is failing with seems to be an issue with Poshi so you can disregard that and I'll make a separate ticket for that. You can patch the warnings if you want it shouldn't make a difference for the test. Just remove that commit and I'll pass this PR for QA and it should be good to forward since there were no other issues.

**Comment by javalosjr96** [comment]
@Kvale1733 
I'll remove the commit.

**Comment by Kvale1733** [comment]
@javalosjr96 QA Passed.

File: modules/apps/batch-engine/batch-engine-service/bnd.bnd [hunk 1]
```diff
@@ -4,8 +4,18 @@ Bundle-Version: 4.0.62
 Import-Package:\
 	!com.graphbuilder.*,\
 	\
+	!de.rototor.pdfbox.graphics2d.*,\
```
Comment (tinatian) [inline modules/apps/batch-engine/batch-engine-service/bnd.bnd:7]: @javalosjr96 I do not think this is something I can get by "Auto SF" according to your commit message. Could you please separate the changes you made manually?

Comment (javalosjr96) [reply modules/apps/batch-engine/batch-engine-service/bnd.bnd:7]: Ok.

Comment (javalosjr96) [reply modules/apps/batch-engine/batch-engine-service/bnd.bnd:7]: @tinatian Commit was split.

**Comment by Kvale1733** [comment]
@tinatian The PR is good to forward from a QA perspective.

File: modules/apps/dynamic-data-lists/dynamic-data-lists-test/build.gradle [hunk 2]
```diff
@@ -9,11 +9,13 @@ dependencies {
...
+	testIntegrationCompile group: "com.liferay", name: "poi", version: "5.2.2.LIFERAY-PATCHED-1"
+	testIntegrationCompile group: "com.liferay", name: "poi-ooxml", version: "5.2.2.LIFERAY-PATCHED-1"
+	testIntegrationCompile group: "com.liferay", name: "poi-ooxml-lite", version: "5.2.2.LIFERAY-PATCHED-1"
+	testIntegrationCompile group: "com.liferay", name: "xmlbeans", version: "5.0.3.LIFERAY-PATCHED-1"
 	testIntegrationCompile group: "com.liferay.portal", name: "com.liferay.portal.impl", version: "default"
 	testIntegrationCompile group: "javax.portlet", name: "portlet-api", version: "3.0.1"
-	testIntegrationCompile group: "org.apache.poi", name: "poi", version: "4.1.2"
-	testIntegrationCompile group: "org.apache.poi", name: "poi-ooxml", version: "4.1.2"
-	testIntegrationCompile group: "org.apache.poi", name: "poi-ooxml-schemas", version: "4.1.2"
+	testIntegrationCompile group: "org.apache.commons", name: "commons-collections4", version: "4.4"
```
Comment (tinatian) [inline modules/apps/dynamic-data-lists/dynamic-data-lists-test/build.gradle:18]: @javalosjr96 This can not be a change caused by AUTO SF

**Comment by tinatian** [comment]
@javalosjr96 Please check my comment. For "AUTO SF", please make sure you run SF locally and commit the changes. Also, for  this commit "LPS-163136 Manually update versions-complete.xml and versions.html", please do not modify it manually, please run "ant build-lib-versions" to regenerate.

Thanks.
Tina

**Comment by javalosjr96** [comment]
ci:test:sf

**Comment by javalosjr96** [comment]
ci:test:relevant

**Comment by javalosjr96** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
No regressions found.

**Pull request closed without merging**